programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -429,7 +461,14 @@ impl<'a, 'info> ScanningAccountRetriever<'a, 'info> {
                     && serum3_cpi::has_serum_header(&x.data.borrow())
             })
             .count();
-        let fallback_oracles_start = serum3_start + n_serum3;
+        let openbook_v2_start = serum3_start + n_serum3;
+        let n_openbook_v2 = ais[openbook_v2_start..]
+            .iter()
+            .take_while(|x| {
+                x.data_len() == std::mem::size_of::<openbook_v2::state::OpenOrdersAccount>() + 8
```

Comment:
just using the length isn't sufficiently safe against accidents - maybe a fallback oracle happens to have the same data length. Should at least check the discriminator. We would love to check the owner, but if it's like obv1, then there could be different programs?

Reply:
added discriminator check so it's at least the same as v1, yeah possible there could be more than one deployment, could have a list of whitelisted program ids maybe